### PR TITLE
Import modules

### PR DESCRIPTION
**Pull request by dfunckt**
Merge a few dependencies together into one tree to provide complete coverage for the use case: given a docker-compose.yml, build the services, and (optionally) create a release for it at balenaCloud.

Clients that are looking to build docker-compose files typically have to depend on and import *all* of these modules, and some of the modules themselves depend on each other creating a situation where maintaining the correct versions of things is very cumbersome.

This commit is the first step to a standalone module that is reasonably useful but currently makes no effort to rethink the exposed APIs. Imported modules are exported as-is. In the future, we should improve by exporting more meaningful APIs.

The merged dependencies are:

- [balena-multibuild](https://github.com/balena-io-modules/balena-multibuild) @ aefcb3248ed183
- [balena-compose-parse](https://github.com/balena-io-modules/balena-compose-parse) @ 06a6b10d3096e
- [balena-release](https://github.com/balena-io-modules/balena-release) @ 83155848aa4c
- [docker-qemu-transpose](https://github.com/balena-io-modules/docker-qemu-transpose) @ 734d8397dfe
- [dockerfile-template](https://github.com/balena-io-modules/dockerfile-template) @ 3ae77d4ec75
- [resin-bundle-resolve](https://github.com/balena-io-modules/resin-bundle-resolve) @ 7a972ce731e5ec4
- [resin-docker-build](https://github.com/balena-io-modules/resin-docker-build) @ 703653cc0eaa57

### CODE REVIEW

**Comment by dfunckt** [comment]
https://github.com/balena-io-modules/docker-qemu-transpose is another one that could be merged in this.

EDIT: Done.

File: lib/multibuild/build.ts [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+	// * _ Does the Docker version support it? _
+	//   NOTE: docker api version 1.34 actually introduced platform to the
+	//         api but it was broken until fixed in 1.38
+	//         https://github.com/moby/moby/commit/7f334d3acfd7bfde900e16e393662587b9ff74a1
+	//
+	// * _ Do the images support it? _
+	//   In order to support `platform`, all images used in the Dockerfile must use the same
+	//   architecture.  Determining this is problematic.  See comments in getRecommendedPlatformHandling
+
+	const usePlatformOption: boolean =
```
Comment (toochevere) [inline lib/multibuild/build.ts:131]: FYI:  Once this merges, in order to unblock the (formerly) multibuild issue, I'll PR a change to allow the default `--platform` behavior here with a new option on the build task `task.useDefaultPlatformForMultiarchBaseImages` and then 

```
	const usePlatformOption: boolean =
		!!task.dockerPlatform &&
		semver.satisfies(
			semver.coerce((await docker.version()).ApiVersion) || '0.0.0',
			'>=1.38.0',
		) &&
		(task.useDefaultPlatformForMultiarchBaseImages ||
			(await checkAllowDockerPlatformHandling(task, docker)));
```

**Comment by dfunckt** [comment]
Also added https://github.com/balena-io-modules/dockerfile-template

**Comment by dfunckt** [comment]
@balena-ci I self-certify!

**Pull request merged**